scripts/install/setup.sh

```diff
@@ -291,10 +289,14 @@ else
   bold "Using existing audit log cloud function $CLOUD_FUNCTION_NAME..."
 fi
 
-# We want the local hal config to match what was deployed.
-~/spinnaker-for-gcp/scripts/manage/pull_config.sh
-# We want a full backup stored in the bucket and the full deployment config stored in a secret.
-~/spinnaker-for-gcp/scripts/manage/push_config.sh
+if [ "$USE_CLOUD_SHELL_HAL_CONFIG" = true ] ; then
+  ~/spinnaker-for-gcp/scripts/manage/push_and_apply.sh
```

Comment:
Another option is to use `gcloud compute disks` to create a persistent disk and copy the hal folder into it prior to the kubectl apply. quick-install.yml would be updated to include a PersistentVolume that would use the predefined disk. This would allow the halyard-daemon container to read the restored hal folder from the volume. I avoided this approach because removing the responsibility of creating a PersistentVolume from Kubernetes and replacing it with direct management of the volume would introduce greater risk and complexity.

Reply:
I prefer the current approach as well.